application/views/ride_views.py

```diff
@@ -133,6 +133,32 @@ def get(self):
             raise e
 
 
+class SingleRide(Resource):
+
+    @api.doc('Get Available rides',
+             params={'ride_id': 'Id for a single ride offer'},
+             responses={200: 'OK', 404: 'NOT FOUND'})
+    @jwt_required
+    def get(self, ride_id):
+        """Retrieves all available rides"""
+        try:
+            query = "SELECT * from rides where rides.ride_id = '{}'"\
+                . format(ride_id)
+            cursor.execute(query)
+            rows = cursor.fetchall()
+
+            if len(rows) > 0:
+                return jsonify([
+                    {'id': row[0], 'start point': row[2],
+                     'destination': row[3], 'start_time': row[4],
+                     'route': row[5],
+                     'available space': row[6]}
+                    for row in rows])
+            return {'message': 'Offer not found'}, 404
+        except Exception as e:
```

Comment:
If I may ask, are you handling responses for exceptions or the client gets the raised exception? I think it's a security issue for the client to know **exactly** what exception was raised. Also, I think Catching too general exception Exception (broad-except) is not a good idea :thinking:?

Reply:
Thanks for the insight. I would take into consideration to catch particular exception instead of the general exception.